### DISCUSSION TITLE
My Prototype Board

### DISCUSSION THREAD

**Discussion by rooftopsolarsa** [ArminJo/JK-BMSToPylontechCAN#27] [Show and tell]
Hey ArminJo
Your project works so well I have decided to also try do a prototype pcb to make it more permanent for my battery banks :)
Thanks to [paulsteigel](https://github.com/paulsteigel) for the inspiration to take this a step further.

I have added RJ45 ports and the ability to use or not use the LCD and buttons on the board.
For the comm ports I am also thinking of adding 2 comms status leds to visually show if comms are working on the BMS side and the CAN side.

[https://easyeda.com/editor#project_id=3af887e6b1f24589b87f126d5ee47b87](https://easyeda.com/editor#project_id=3af887e6b1f24589b87f126d5ee47b87)

![jkbms-can-coms-rev102-circuit-view](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/153637794/8bb41ca5-5893-441a-9d97-f85ff1e97c43)

![jkbms-can-coms-rev102-top-view](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/153637794/6a7c8a91-392f-4f2b-b375-128acc52b9b9)

![jkbms-can-coms-rev102-bottom-view](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/153637794/2577a029-fbf9-4a56-9999-4fecd3c28368)

**Comment by ArminJo** [comment]
How do you attach the [CAN Board](https://de.aliexpress.com/item/32799974260.html)? 
From below? 
I would suggest to attach it from top but upside down (after removing the J2 301 connectors). 
But then your CAN module part must be mirrored.

**Reply by rooftopsolarsa** [reply]
I am still refining the prototype pcb :)
My CAN module does not have power connector and pin headers, so for me I can simply solder it as-is with my own pin headers direct into position.
I have also discarded the leds and changed the rj45 connectors to use with builtin leds.

![jkbms-can-coms-rev104-top-view](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/153637794/bef2d4af-1860-4553-97ba-0580918b1b28)

**Reply by ArminJo** [reply]
@rooftopsolarsa 
Hi,
I linked your design as an example in the Readme. Do you have a version, where the status LEDs are used?
Until now I see it on the PCD, but in the schematics, they are missing.

**Comment by paulsteigel** [comment]
Hi rooftopsolarsa,
Why dont you use header for the nano, it's better for take it out/in when needed!
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/a2dab58c-5d1a-4f92-8ade-185b34db5244)
JK-BMS has quite some problems in calculating SOC after sometime. I finally have to make some modifications from Armin code to remap SOC (sending fake soc to the inverter for it to work better) using voltage and also inject CAN message to stop charging rather than waiting the inverter to stop charging. Also, at fully charge, we will have to actively stop sending the additional charging request to the inverter to avoid further bulk charging.
However, Armin works is great and helpful for us!
All the best and happy new year to you!
Ngoc
